### PR TITLE
EZP-24800 Support allowed classes limitation for object relation fields

### PR DESCRIPTION
**Pull request by slaci**
This is related to this PR: https://github.com/ezsystems/ezpublish-kernel/pull/2022

I made a base class for the relation forms instead of duplication. I hope this is not a problem.

It adds a new translation... dunno whats the proper way for that. I updated the file by hand.

### CODE REVIEW

**Comment by bdunogier** [comment]
Thank you for this, i'll have a look tomorrow !

As for translations, you can run bin/extract-translations.sh from the repository-forms directory.

**Comment by andrerom** [comment]
+1, but 'I'll let the reviewers give the proper review :)

**Comment by slaci** [comment]
> As for translations, you can run bin/extract-translations.sh from the repository-forms directory.
Thanks for the tip. I ran this command and it updated some line references and moved my new entry to an other line... so I let it happen. I had to move the `configureOptions` method into both children classes, cause the extractor does not handle parent classes (at least for this), so it created a messages.en.xlf without this.

**Comment by slaci** [comment]
I created a separate commit for the translation file. Nothing else changed (I hope 🙈 )

File: lib/FieldType/Mapper/AbstractRelationFormMapper.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        $this->contentTypeService = $contentTypeService;
+        $this->translationHelper = $translationHelper;
+    }
+
+    /**
+     * Fill a hash with all content types and their ids.
+     * @return array
+     */
+    protected function getContentTypeHash()
```
Comment (bdunogier) [inline lib/FieldType/Mapper/AbstractRelationFormMapper.php]: Plural, `getContentTypesHash()`.

Comment (andrerom) [inline lib/FieldType/Mapper/AbstractRelationFormMapper.php]: @slaci can you fix this nitpic? :)

**Comment by andrerom** [comment]
I'll handle nitpick as followup commit, thanks @slaci.

**Pull request merged**